File: src-all/Python/S0282-expression-add-operators.py
"""
Given a string that contains only digits 0-9 and a target value, return all possibilities to add binary operators (not unary) +, -, or * between the digits so they evaluate to the target value.

Example 1:
Input: num = &quot;123&quot;, target = 6
Output: [&quot;1+2+3&quot;, &quot;1*2*3&quot;] 

Example 2:
Input: num = &quot;232&quot;, target = 8
Output: [&quot;2*3+2&quot;, &quot;2+3*2&quot;]

Example 3:
Input: num = &quot;105&quot;, target = 5
Output: [&quot;1*0+5&quot;,&quot;10-5&quot;]

Example 4:
Input: num = &quot;00&quot;, target = 0
Output: [&quot;0+0&quot;, &quot;0-0&quot;, &quot;0*0&quot;]

Example 5:
Input: num = &quot;3456237490&quot;, target = 9191
Output: []

 
Constraints:
	0 <= num.length <= 10
	num only contain digits.


"""
from typing import List
class Solution:
    def addOperators(self, num: str, target: int) -> List[str]:
        
        pass


if __name__ == '__main__':
    assert Solution().addOperators(0) == 0

File: src-all/C++/S0282-expression-add-operators.cpp
/*
Given a string num that contains only digits and an integer target, return all possibilities to insert the binary operators '+', '-', and/or '*' between the digits of num so that the resultant expression evaluates to the target value.

Note that operands in the returned expressions should not contain leading zeros.

 
Example 1:
Input: num = "123", target = 6
Output: ["1*2*3","1+2+3"]
Explanation: Both "1*2*3" and "1+2+3" evaluate to 6.

Example 2:
Input: num = "232", target = 8
Output: ["2*3+2","2+3*2"]
Explanation: Both "2*3+2" and "2+3*2" evaluate to 8.

Example 3:
Input: num = "3456237490", target = 9191
Output: []
Explanation: There are no expressions that can be created from "3456237490" to evaluate to 9191.

 
Constraints:
	1 <= num.length <= 10
	num consists of only digits.
	-231 <= target <= 231 - 1


*/
#include <cassert>
#include <iostream>

class Solution {
public:
    vector<string> addOperators(string num, int target) {
        
    }
};

int main() {
  Solution sol;
  assert(sol.addOperators(0) == 0);
  std::cout << "Pass test cases!" << std::endl;
}